exporter/prometheusremotewriteexporter/exporter.go

```diff
@@ -238,6 +278,11 @@ func (prwe *PRWExporter) export(ctx context.Context, tsMap map[string]*prompb.Ti
 			defer wg.Done()
 
 			for request := range input {
+				if err := ctx.Err(); err != nil {
+					// TODO(@odeke-em): Perhaps log this error.
+					_ = err
```

Comment:
Perhaps logger.Debug() log if unsure that it can happen frequently and flood the logs, but still important for troubleshooting?

Reply:
So that would have involved adding a logger to the system but we can actually just check that the context was cancelled, so I've done exactly that in a select. Thank you @tigrannajaryan, please take a look at the newest update.